parsers/lib/validation.py

```diff
@@ -27,6 +31,46 @@ def check_expected_range(datapoint, value, expected_range, logger, key=None):
     return True
 
 
+def validate_diffs(datapoints: list, max_diff: dict, logger: logging.Logger):
+    """
+
+    Parameters
+    ----------
+    datapoints: a list of datapoints having a 'production' field
+    max_diff: dict representing the max allowed diff per energy type
+    logger
+
+    Returns
+    -------
+    the same list of datapoints, with the ones having a too big diff removed
+    """
+    # sort datapoins by datetime
+    datapoints = sorted(datapoints, key=lambda x: x['datetime'])
+
+    ok_diff = np.ones_like(datapoints, dtype=bool)
+    for energy, max_diff in max_diff.items():
+        series = pd.Series(
+            [datapoint['production'].get(energy, np.nan)
+             for datapoint in datapoints])
+        # nan is always allowed (can be disallowed using `validate` function)
+        new_diffs = (np.abs(series.diff()) < max_diff) | series.isna()
+        if not new_diffs[1:].all():
+            wrongs_ixs = new_diffs[~new_diffs].index
+            wrongs_ixs_and_previous = sorted(
+                {ix - 1 for ix in wrongs_ixs} | set(wrongs_ixs))
+            to_display = [
+                (datapoints[i]['datetime'], datapoints[i]['production'][energy])
+                for i in wrongs_ixs_and_previous if i > 0]
+            logger.warning(
+                f'some datapoints have a too high diff for {energy}: '
```

Comment:
can we rename `diff` to `time difference` (or something alike)? It makes the error more explicit.

Reply:
yes, we can make it more explicit, but it's the absolute difference in the production value we're testing here, not the time difference

Reply:
I'm not sure I understand: aren't you comparing the difference between production at time t and production at time t-1 for a specific production mode?